modules/@apostrophecms/schema/ui/apos/components/AposSchema.vue

```diff
@@ -95,6 +95,14 @@ export default {
       handler() {
         this.updateNextAndEmit();
       }
+    },
+    value: {
+      deep: true,
+      handler(newVal, oldVal) {
+        if (newVal.data.slug !== this.fieldState.slug.data) {
```

Comment:
Not every schema has a slug field at all, for example relationships that have schemas. This is hardcoding a special case in a way that will crash if there isn't a slug. I'm also not sure why it's being done.

Reply:
In the original PoC components, slug fields were set up to watch title fields without polluting ApostropheSchema with special-case code. This was done via the `context` prop passed to all fields and implemented by the ApostropheFieldMixin. It looks like this feature got lost in the new Vue components.

Reply:
(I'm not sure if this code is related to slugs watching titles, or something else.)

Reply:
I don't know if the `context` bit is related to this. The situation here is that in the image editor the document in the editor might be completely swapped out for another image document at any time. The schema needs to recognize this and update the field values.

Once the doc is passed into `AposSchema` it is updated to reflect the schema fields. This means that `_id` is removed, among others. So the only unique field I had left was `slug`.

Reply:
I see. Wouldn't we always want to refresh the whole thing when any part of `value` is modified, not just slug? That's how the v-model pattern works: watch changes to the `value` prop to accept change from above, emit the `input `event to push change from below.